### PR TITLE
fix(Tag): ignore click events when disabled

### LINKED ISSUE #4791: Filter tag fires click event when disabled
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

When clicking on the filter tag close icon, the onclick event is still dispatched even if the component is disabled

> Is this issue related to a specific component?

filter tag

> What did you expect to happen? What happened instead? What would you like to
> see changed?

the click event should not propagate from the disabled filter tag

## Steps to reproduce the issue

1. view the filter tag story
2. disabled the component
3. click on the component and view the story actions panel

### PR DESCRIPTION
**Pull request by emyarod**
Closes #4791

~~This PR wraps the close icon in a button~~ to utilize native button features (rather than spoofing a button with the SVG element) and ignores the onClick handler when the filter tag is disabled

#### Changelog

**New**

- extend disabled tag styles for filter tag

**Changed**

- ~~wrap SVG icon in button~~ due to changes introduced in #4863 the entire tag is now a `button` and the styles are rescoped to the svg icon within the `button.bx--tag` element
- click handler logic
- update test to remove console warning

#### Testing / Reviewing

Ensure click handlers are no longer fired when the tag filter is disabled and the component matches the visual spec (particularly the disabled state)

### CODE REVIEW

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

File: packages/react/src/components/Tag/Tag.js [hunk 1]
```diff
@@ -41,13 +42,14 @@ const Tag = ({
...
-      tabIndex="0" // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
-      {...other}>
+    <span className={tagClasses} {...other}>
       {children !== null && children !== undefined ? children : TYPES[type]}
-      <Close16 aria-label={title || 'Clear filter'} />
+      <button
+        className={`${prefix}--tag--filter__close-button`}
+        disabled={disabled}
+        onClick={!disabled ? onClick : undefined}>
+        <Close16 aria-label={title || 'Clear filter'} />
```
Comment (joshblack) [inline packages/react/src/components/Tag/Tag.js]: Would we want to place `aria-label` on the button instead?

**Review by joshblack — APPROVED** [review]
Seems great to me, just a quick comment

**Review by aledavila — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Review by laurenmrice — APPROVED** [review]
disabled tag looks good to me and is not clickable 👍

File: packages/components/src/components/tag/tag.hbs [hunk 2]
```diff
@@ -15,9 +15,9 @@
 
 <div role="list">
   {{#if filter}}
-  <span class="{{@root.prefix}}--tag {{@root.prefix}}--tag--filter" title="Clear filter" tabindex="0" role="listitem">
+  <button class="{{@root.prefix}}--tag {{@root.prefix}}--tag--filter" title="Clear filter" role="listitem">
```
Comment (joshblack) [inline packages/components/src/components/tag/tag.hbs]: Does `role="listitem"` make sense if it's now a button? 🤔 Would we want to handle hbs updates in a separate PR?

File: packages/components/src/components/tag/_tag.scss [hunk 3]
```diff
@@ -97,31 +98,42 @@
...
     box-shadow: inset 0 0 0 2px $inverse-focus-ui;
     border-radius: 50%;
   }
 
+  .#{$prefix}--tag--filter.#{$prefix}--tag--disabled svg:hover {
+    background-color: transparent;
+  }
+
+  .#{$prefix}--tag--filter.#{$prefix}--tag--disabled svg {
+    fill: $ibm-color__gray-30;
```
Comment (joshblack) [inline packages/components/src/components/tag/_tag.scss]: Is there no token for this? Surprised we had to dip into IDL colors, if we have to use them we might want to switch to the carbon namespaced ones as the ibm ones will be removed.

File: packages/react/src/components/Tag/Tag.js [hunk 4]
```diff
@@ -48,8 +49,12 @@ const Tag = ({
           ? `${title} ${children}`
           : `Clear filter ${children}`
       }
+      disabled={disabled}
+      onClick={!disabled ? onClick : undefined}
```
Comment (joshblack) [inline packages/react/src/components/Tag/Tag.js]: Does a `button` fire an `onClick` event if it's `disabled`?

**Review by joshblack — CHANGES_REQUESTED** [review]

File: packages/components/src/components/tag/_tag.scss (hunk 3 above)
Comment (emyarod) [reply packages/components/src/components/tag/_tag.scss]: cc @laurenmrice for the required token here (the filter tag close 'x' icon color when disabled)

File: packages/components/src/components/tag/tag.hbs (hunk 2 above)
Comment (emyarod) [reply packages/components/src/components/tag/tag.hbs]: removed the outdated roles. it's a documentation change so I think it's ok for it to be contained in this PR

File: packages/react/src/components/Tag/Tag.js (hunk 4 above)
Comment (emyarod) [reply packages/react/src/components/Tag/Tag.js]: it doesn't, I think that was left over from a merge conflict resolution

File: packages/components/src/components/tag/_tag.scss (hunk 3 above)
Comment (emyarod) [reply packages/components/src/components/tag/_tag.scss]: confirmed token value offline

**Comment by tw15egan** [comment]
@joshblack is this good to go now?

**Review by joshblack — APPROVED** [review]

**Pull request merged**